### PR TITLE
Create Transpiler Foundation

### PR DESCRIPTION
**Pull request by gregbartell**
Primary modification: add classes.py and main.py to create a base to write the transpiler on.

Additionally, I added some matching rules regarding my personal config files (necessary for autocomplete in my setup)

### CODE REVIEW

File: transpile/classes.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/python3
+
+class Block:
```
Comment (mpoegel) [inline transpile/classes.py]: This should be defined as an abstract base class (see the [docs](https://docs.python.org/3/library/abc.html))

File: transpile/classes.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    id = -1
```
Comment (mpoegel) [inline transpile/classes.py]: You're defining all of these as static variables I believe which is not what should be done. They should be member variables.
```python
class Block(object):
    def __init__(self):
        self._id = -1
        # etc.
```

Same goes for the other classes.

**Comment by gregbartell** [comment]
@mpoegel

http://pastebin.com/Zd0TBkck

I have no idea what I'm doing here. Tell me if this is sort of what you're going for. In particular, the Abstract Base Class thing makes no sense to me.

**Comment by droghio** [comment]
@gregbartell The idea of an abstract base class is that you have a "mytic" block class that all other blocks inherit from. That said no block can channel the pure blockiness of the base class, so it is a mythical abstraction i.e. abstract class. All attempts to generate a pure block should be protected/prevented as the computer would be unable to cope with its pure block-as-ity.

**Comment by droghio** [comment]
That said I why are you emulating the frontend code. Do you really need a class representation of the JSON file, or would it be enough to just iterator through a dict equivalent and essentially use find/replace to generate code for the Arudino?

**Comment by mpoegel** [comment]
You guys should stop switching between single quotes and double quotes. Most style guides call for single quotes everywhere.

**Comment by gregbartell** [comment]
@mpoegel I don't know what our style guide says, but I've always been taught that in Python you should use double quotes for strings the user might see (ex: print("Hello, World!")), and single for everything else (ex: open('file.txt', 'r')).

I'll switch to whatever you guys want though, I don't feel strongly about it. Functionally, they're exactly equivalent (unlike C-style languages).

**Pull request merged**